cmd/podman/shared/container.go

```diff
@@ -64,6 +94,201 @@ type ContainerSize struct {
+		if err != nil {
+			return errors.Wrapf(err, "unable to obtain container state")
+		}
+		command = strings.Join(c.Command(), " ")
+		created = units.HumanDuration(time.Since(c.CreatedTime())) + " ago"
+
+		exitCode, _, err = c.ExitCode()
+		if err != nil {
+			return errors.Wrapf(err, "unable to obtain container exit code")
+		}
+		startedAt, err = c.StartedTime()
+		if err != nil {
+			logrus.Errorf("error getting started time for %q: %v", c.ID(), err)
+		}
+		exitedAt, err = c.FinishedTime()
+		if err != nil {
+			logrus.Errorf("error getting exited time for %q: %v", c.ID(), err)
+		}
+		if opts.Namespace {
+			pid, err = c.PID()
+			if err != nil {
+				return errors.Wrapf(err, "unable to obtain container pid")
+			}
+		}
+		if opts.Size {
+			size = new(ContainerSize)
+
+			rootFsSize, err := c.RootFsSize()
+			if err != nil {
+				logrus.Errorf("error getting root fs size for %q: %v", c.ID(), err)
+			}
+
+			rwSize, err := c.RWSize()
+			if err != nil {
+				logrus.Errorf("error getting rw size for %q: %v", c.ID(), err)
+			}
+
+			size.RootFsSize = rootFsSize
+			size.RwSize = rwSize
+		}
+
+		return nil
+	})
+
+	if batchErr != nil {
+		return PsContainerOutput{}, batchErr
+	}
+
+	switch conState.String() {
+	case libpod.ContainerStateExited.String():
+		fallthrough
+	case libpod.ContainerStateStopped.String():
+		exitedSince := units.HumanDuration(time.Since(exitedAt))
+		status = fmt.Sprintf("Exited (%d) %s ago", exitCode, exitedSince)
+	case libpod.ContainerStateRunning.String():
+		status = "Up " + units.HumanDuration(time.Since(startedAt)) + " ago"
+	case libpod.ContainerStatePaused.String():
+		status = "Paused"
+	case libpod.ContainerStateCreated.String(), libpod.ContainerStateConfigured.String():
+		status = "Created"
+	default:
+		status = "Error"
+	}
+
+	_, imageName := ctr.Image()
+	cid := ctr.ID()
+	pod := ctr.PodID()
+	if !opts.NoTrunc {
+		cid = cid[0:cidTruncLength]
+		if len(pod) > 12 {
+			pod = pod[0:podTruncLength]
+		}
+	}
+
+	return PsContainerOutput{
+		ID:        cid,
+		Image:     imageName,
+		Command:   command,
+		Created:   created,
+		Ports:     portsToString(ctr.PortMappings()),
+		Names:     ctr.Name(),
+		IsInfra:   ctr.IsInfra(),
+		Status:    status,
+		State:     conState,
+		Pid:       pid,
+		Size:      size,
+		Pod:       pod,
+		ExitedAt:  exitedAt,
+		CreatedAt: ctr.CreatedTime(),
+		StartedAt: startedAt,
+		Labels:    ctr.Labels(),
+	}, nil
+	return PsContainerOutput{}, nil
+
+}
+
+type pFunc func() (PsContainerOutput, error)
+
+type workerInput struct {
+	parallelFunc pFunc
+	opts         PsOptions
+	cid          string
+	job          int
+}
+
+// worker is a "threaded" worker that takes jobs from the channel "queue"
+func worker(wg *sync.WaitGroup, jobs <-chan workerInput, results chan<- PsContainerOutput, errors chan<- error) {
+	for j := range jobs {
+		r, err := j.parallelFunc()
+		// If we find an error, we return just the error
+		if err != nil {
+			errors <- err
+			// Return the result
```

Comment:
Should this be two lines down?

Reply:
I don't believe so; the design we landed on is we only "return" the error if it is non-nil; otherwise we return the results so that in the end we just check if there are errors in the channel.